tests/cupy_tests/fft_tests/test_fft.py

```diff
@@ -359,31 +368,36 @@ def test_fft_allocate(self):
         cupy.get_default_memory_pool().free_all_blocks()
 
 
-@testing.parameterize(
-    {'shape': (3, 4), 's': None, 'axes': None, 'norm': None},
-    {'shape': (3, 4), 's': (1, None), 'axes': None, 'norm': None},
-    {'shape': (3, 4), 's': (1, 5), 'axes': None, 'norm': None},
-    {'shape': (3, 4), 's': None, 'axes': (-2, -1), 'norm': None},
-    {'shape': (3, 4), 's': None, 'axes': (-1, -2), 'norm': None},
-    {'shape': (3, 4), 's': None, 'axes': (0,), 'norm': None},
-    {'shape': (3, 4), 's': None, 'axes': None, 'norm': 'ortho'},
-    {'shape': (3, 4), 's': None, 'axes': (), 'norm': None},
-    {'shape': (2, 3, 4), 's': None, 'axes': None, 'norm': None},
-    {'shape': (2, 3, 4), 's': (1, 4, None), 'axes': None, 'norm': None},
-    {'shape': (2, 3, 4), 's': (1, 4, 10), 'axes': None, 'norm': None},
-    {'shape': (2, 3, 4), 's': None, 'axes': (-3, -2, -1), 'norm': None},
-    {'shape': (2, 3, 4), 's': None, 'axes': (-1, -2, -3), 'norm': None},
-    {'shape': (2, 3, 4), 's': None, 'axes': (0, 1), 'norm': None},
-    {'shape': (2, 3, 4), 's': None, 'axes': None, 'norm': 'ortho'},
-    {'shape': (2, 3, 4), 's': None, 'axes': (), 'norm': None},
-    {'shape': (2, 3, 4), 's': (2, 3), 'axes': (0, 1, 2), 'norm': 'ortho'},
-    {'shape': (2, 3, 4, 5), 's': None, 'axes': None, 'norm': None},
-    {'shape': (0, 5), 's': None, 'axes': None, 'norm': None},
-    {'shape': (2, 0, 5), 's': None, 'axes': None, 'norm': None},
-    {'shape': (0, 0, 5), 's': None, 'axes': None, 'norm': None},
-    {'shape': (3, 4), 's': (0, 5), 'axes': None, 'norm': None},
-    {'shape': (3, 4), 's': (1, 0), 'axes': None, 'norm': None},
-)
+@pytest.mark.usefixtures('skip_forward_backward')
+@testing.parameterize(*(
+    testing.product_dict([
+        {'shape': (3, 4), 's': None, 'axes': None},
+        {'shape': (3, 4), 's': (1, None), 'axes': None},
+        {'shape': (3, 4), 's': (1, 5), 'axes': None},
+        {'shape': (3, 4), 's': None, 'axes': (-2, -1)},
+        {'shape': (3, 4), 's': None, 'axes': (-1, -2)},
+        {'shape': (3, 4), 's': None, 'axes': (0,)},
+        {'shape': (3, 4), 's': None, 'axes': None},
```

Comment:
The triple (shape, s, axes) is repeated.

By the way, testing all `norm`s for each triple may be heavy.  Could you compare the numbers/time of tests between before and after the PR?

Reply:
Sorry, you meant repeated with different `norm`s, right?

I checked that the number of tests is doubled (2 `norm`s -> 4 `norm`s, if testing `norm=''` then 5), but the increased time is negligible as FFT is very fast.

Reply:
@toslunar I found a weird NumPy behavior which I think is a bug that should be reported and fixed upstream:
```python
>>> import numpy as np
>>>
>>> a = np.random.random((2,3,4))
>>> b = np.fft.fft2(a, norm='')  # <-- this does not work (norm invalid)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 6, in fft2
  File "/home/leofang/miniconda3/envs/cupy_cuda112_dev/lib/python3.7/site-packages/numpy/fft/_pocketfft.py", line 1012, in fft2
    return _raw_fftnd(a, s, axes, fft, norm)
  File "/home/leofang/miniconda3/envs/cupy_cuda112_dev/lib/python3.7/site-packages/numpy/fft/_pocketfft.py", line 705, in _raw_fftnd
    a = function(a, n=s[ii], axis=axes[ii], norm=norm)
  File "<__array_function__ internals>", line 6, in fft
  File "/home/leofang/miniconda3/envs/cupy_cuda112_dev/lib/python3.7/site-packages/numpy/fft/_pocketfft.py", line 214, in fft
    inv_norm = _get_forward_norm(n, norm)
  File "/home/leofang/miniconda3/envs/cupy_cuda112_dev/lib/python3.7/site-packages/numpy/fft/_pocketfft.py", line 88, in _get_forward_norm
    raise ValueError(f'Invalid norm value {norm}; should be "backward",'
ValueError: Invalid norm value ; should be "backward","ortho" or "forward".
>>> b = np.fft.fft2(a, norm='', axes=())  # <-- this works (despite norm is invalid)?!
>>> (b == a).all()
True
```
If `axes=()` then `norm` is not checked (NumPy just returns the input array). Should I just skip testing this corner case?

Reply:
> If `axes=()` then `norm` is not checked (NumPy just returns the input array). Should I just skip testing this corner case?

Workaround done in f8bd913.